pbf/models.py

```diff
@@ -218,10 +218,36 @@ class Meta:
     # It would be best to rename starting_energy to base_energy_per_turn,
     # but this will need a database change.
     starting_energy = models.IntegerField(default=0)
+    # A number of spirits have a resource that's specific to them.
+    # Rather than have separate fields + endpoints that can modify each of them,
+    # we'll use a single field for this purpose,
+    # with the main motivation of avoiding repetitive code.
+    # (and a secondary motivation of having fewer columns in the database)
+    #
+    # To see the list of spirits that use this field, see spirit_specific_resource_name below.
+    spirit_specific_resource = models.IntegerField(default=0)
+
+    rot_gained_this_turn = models.BooleanField(default=False)
+    rot_converted_this_turn = models.BooleanField(default=False)
```

Comment:
I'd like to think more about adding these two boolean fields. There is an alternative design, where we use the lowest two bits of the spirit specific resource field to store whether rot has been gained/converted this turn. That way we don't have to keep adding boolean fields for every single spirit that might need one.

I came up with this when thinking about https://github.com/nathanj/spirit-island-pbp/issues/52, where Dances Up Earthquakes might have a spirit-specific boolean (whether impending energy has been gained this turn).

I'd like to explore more about this idea, so I'd like to withdraw Spreading Rot Renews the Earth from review consideration for the time being, please. I will keep Covets Gleaming Shards of Earth in review consideration since that does not require a spirit-specific boolean.

Reply:
Thinking about this a bit more, I think it gets a little hard to understand the field if it both contains a value that can be incremented and a number of flags. I think It'd be reasonable to have one `spirit_specific_resource` (acts as before), and one `spirit_specific_per_turn_flags`.